app/controllers/webhooks_controller.rb

```diff
@@ -120,55 +123,87 @@ def handle_payment_intent_succeeded(
 
       amount = item['amount']
       seller_id = item['seller_id']
-      merchant_name = Seller.find_by(seller_id: seller_id).name
-      case item['item_type']
-      when 'donation'
-        item = create_item(
-          item_type: :donation,
-          seller_id: seller_id,
-          purchaser: purchaser,
-          payment_intent: payment_intent
-        )
-        create_donation(item: item, amount: amount)
-        begin
-          send_donation_receipt(
-            payment_intent: payment_intent,
-            amount: amount,
-            merchant: merchant_name
-          )
-        rescue StandardError
+
+
+      if seller_id.eql?(POOL_DONATION_SELLER_ID)
+        unless item['item_type'].eql?('donation')
+          raise InvalidPoolDonationError,
+                "pool contribution must but be of type 'donation' but found type #{item['item_type']}."
         end
-      when 'gift_card'
-        purchaser = payment_intent.purchaser
-        item = create_item(
-          item_type: :gift_card,
-          seller_id: seller_id,
-          purchaser: purchaser,
-          payment_intent: payment_intent
-        )
 
-        recipient = payment_intent.recipient
-        gift_card_detail = create_gift_card(
-          item: item,
-          amount: amount,
-          seller_id: seller_id,
-          recipient: recipient
-        )
+        # calculate amount per merchant
+        # This will break if we ever have zero merchants but are still
+        # accepting pool donations.
+        amount_per = (amount.to_f/(Seller.count - 1).to_f).round
+
+        Seller.all.each do |seller|
+          unless seller.seller_id.eql?(POOL_DONATION_SELLER_ID)
+            new_item = create_item(
```

Comment:
Line 141 to 147 could be abstracted out and shared with the other donation code-path